### PR TITLE
Upgraded type definitions for ssh2-sftp-client to 2.4

### PR DESCRIPTION
**Pull request by viamuli**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/jyu213/ssh2-sftp-client>
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Note:
This is my first pull request ever. Please notify me if/what I have done wrong.

### CODE REVIEW

File: types/ssh2-sftp-client/index.d.ts [hunk 1]
```diff
@@ -49,4 +55,14 @@ declare namespace sftp {
         owner: number;
         group: number;
     }
+
+    interface FileStats {
```
Comment (ascariandrea) [inline types/ssh2-sftp-client/index.d.ts]: Does than the `FileInfo` extend `FileStats` somehow?

Comment (viamuli) [reply types/ssh2-sftp-client/index.d.ts]: Yes, it actally does though FileInfo does not contain the members 'mode' and 'permissions'. Should I add the inheritance then (with mode and permissions as optional members)? And: Is it even possible to add a new commit to an existing pull request?

Comment (ascariandrea) [reply types/ssh2-sftp-client/index.d.ts]: If there's no real inheritance for `FileStats` from `FileInfo` than is ok to have 2 different interfaces.

Of course you can, just keep committing on this branch and push!

Comment (viamuli) [reply types/ssh2-sftp-client/index.d.ts]: Fine, than this could be merged imo.

**Comment by viamuli** [comment]
So when and how is this going to be merged? Is there any action required from me? I'm tired of the compiler errors I run into because of the missing type definitions.

**Review by ascariandrea — APPROVED** [review]

**Comment by ascariandrea** [comment]
Sorry @viamuli , I had to approve the PR - thought I did.
A maintainer should merge it now, I hope.

**Pull request merged**